scripts/testing/database-emulator-test.js

```diff
@@ -0,0 +1,121 @@
+/**
```

Comment:
Just to double-check, is there a reason this is written in JavaScript rather than TypeScript?  I don't care too much, but I think generally we'd prefer the type-safety and extended featureset of TypeScript for any new code we're writing...

Reply:
Agreed.

Looks like all other file in the 'scripts' are written in JavaScript, so I figured I'd do the same. I can help convert them to TypeScript if I get a chance.

Reply:
Agreed. I can help look into this one.

Reply:
I think [ts-node](https://www.npmjs.com/package/ts-node) does that (and is quite popular).